<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Roboto Mono</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>InLanD - Análisis Léxico y Sintáctico</string>
  </property>
  <property name="styleSheet">
   <string>
    QMainWindow {
        background-color: #2E3440;
    }
    QLabel, QTableWidget, QPlainTextEdit, QTextEdit, QPushButton {
        color: #D8DEE9;
        font-family: 'Roboto Mono';
    }
    QPushButton {
        background-color: #4C566A;
        border: 1px solid #4C566A;
        border-radius: 5px;
        padding: 8px;
        min-width: 80px;
    }
    QPushButton:hover {
        background-color: #5E81AC;
    }
    QTabWidget::pane {
        border: 1px solid #4C566A;
        padding: 5px;
    }
    QTabWidget::tab-bar {
        alignment: center;
    }
    QTabBar::tab {
        background: #3B4252;
        color: #D8DEE9;
        padding: 8px;
        margin: 2px;
    }
    QTabBar::tab:selected {
        background: #4C566A;
        border-bottom: 2px solid #88C0D0;
    }
    QHeaderView::section {
        background-color: #4C566A;
        color: #D8DEE9;
        padding: 4px;
        border: none;
    }
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <!-- Entrada de texto -->
    <item>
     <widget class="QLabel" name="inputLabel">
      <property name="text">
       <string>Texto de Entrada:</string>
      </property>
      <property name="styleSheet">
       <string>color: #88C0D0; font-weight: bold;</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="inputTextEdit">
      <property name="styleSheet">
       <string>
        background-color: #3B4252;
        color: #D8DEE9;
        border-radius: 5px;
        padding: 10px;
       </string>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
      </property>
     </widget>
    </item>

    <!-- Botones -->
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="openButton">
        <property name="text">
         <string>Abrir</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveButton">
        <property name="text">
         <string>Guardar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Limpiar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="analyzeButton">
        <property name="text">
         <string>Analizar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exitButton">
        <property name="text">
         <string>Salir</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Sección de Resultados con pestañas -->
    <item>
     <widget class="QTabWidget" name="resultsTab">
      <property name="styleSheet">
       <string>color: #D8DEE9; background-color: #3B4252; border-radius: 5px;</string>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
      </property>
      <!-- Pestaña Tokens -->
      <widget class="QWidget" name="tokensTab">
       <attribute name="title">
        <string>Tokens</string>
       </attribute>
       <layout class="QVBoxLayout" name="tokensLayout">
        <item>
         <widget class="QTableWidget" name="resultTable">
          <property name="styleSheet">
           <string>background-color: #3B4252; color: #D8DEE9; border-radius: 5px; padding: 5px;</string>
          </property>
          <property name="columnCount">
           <number>2</number>
          </property>

          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
          </property>


         </widget>
        </item>
       </layout>
      </widget>

      <!-- Pestaña Errores -->
      <widget class="QWidget" name="errorsTab">
       <attribute name="title">
        <string>Errores</string>
       </attribute>
       <layout class="QVBoxLayout" name="errorsLayout">
        <item>
         <widget class="QPlainTextEdit" name="errorTextEdit">
          <property name="styleSheet">
           <string>background-color: #3B4252; color: #D8DEE9; border-radius: 5px; padding: 10px;</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>

      <!-- Pestaña Sintaxis -->
      <widget class="QWidget" name="syntaxTab">
       <attribute name="title">
        <string>Sintaxis</string>
       </attribute>
       <layout class="QVBoxLayout" name="syntaxLayout">
        <item>
         <widget class="QPlainTextEdit" name="syntaxTextEdit">
          <property name="styleSheet">
           <string>background-color: #3B4252; color: #D8DEE9; border-radius: 5px; padding: 10px;</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
